kernel/src/context.rs

```diff
@@ -85,16 +85,40 @@ impl<'a> CurrentUserSpace<'a> {
     ///
     /// Returns `Err` if the `vaddr` and `len` do not represent a user space memory range.
     pub fn reader(&self, vaddr: Vaddr, len: usize) -> Result<VmReader<'_, Fallible>> {
+        if len > 0 {
```

Comment:
Can anyone remind me why we want to skip the check when `len == 0`? I can't see the point of allowing use cases such as `let reader = CurrentUserSpace.reader(0, 0)`. It does not seem to do any harm, but it is not doing us any favor, either. In my opinion, a caller doing  `CurrentUserSpace.reader(0, 0)` is likely to be a bug. So reporting errors for such cases seems to be a design decision.

Reply:
@junyang-zh Git blame shows you are the author of this check. Could you shed some light?

Reply:
A further investigation shows that @junyang-zh just moved the code from another file authored by @cchanging . @cchanging do you want to weigh in?